core/trino-spi/src/main/java/io/trino/spi/predicate/Domain.java

```diff
@@ -313,7 +313,19 @@ public String toString()
 
     public String toString(ConnectorSession session)
     {
-        return "[ " + (nullAllowed ? "NULL, " : "") + values.toString(session) + " ]";
+        if (isAll()) {
+            return "[ALL]";
+        }
+        if (isNone()) {
+            return "[NONE]";
+        }
+        if (isOnlyNull()) {
+            return "[NULL]";
+        }
+        if (!nullAllowed) {
+            return values.toString(session);
+        }
+        return "[ NULL, " + values.toString(session) + " ]";
```

Comment:
That looks like it's going to look inconsistent.

Reply:
The brackets in case of non-null were extra, but reverted it to keep consistency